### DISCUSSION TITLE
Deleted Columns Re-appearing in the Report After Data Refresh

### DISCUSSION THREAD

**Discussion by pbiuser-sm** [TabularEditor/TabularEditor3#543] [🙋♀️TE3 Q&A]
Hi,
 
I got a requirement to remove all the unused columns from the Power BI model. To make the changes in bulk I’m using tabular editor scripts. After applying the fixes through tabular editor, changes were reflecting in the PBIX file. But when I do a data refresh, the deleted columns are reappearing in the PBIX. I noticed that no additional steps were created in the M-code after making the changes through tabular editor. Could anyone help me with this?
 
Given below are the steps I followed.
 
1.	From Power BI Desktop, Click on External Tools > Tabular Editor
2.	Executed the script to remove columns in advanced editor.
Sample Code: Model.Tables["TableName"].Columns["Column Name"].Delete();
3.	Click Run
4.	When finished, click “Save Back to Model”
 
Thanks in Advance,

**Comment by otykier** [comment]
Adding/editing/removing columns against a Power BI Desktop data model, is [currently not supported by Microsoft](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#data-modeling-operations). Make sure you have the "Allow unsupported modelling operations" option in Tabular Editor **disabled** to prevent making unsupported changes to the model.

![image](https://user-images.githubusercontent.com/8976200/169762473-e3c14782-8ed4-4ae4-af9e-2ffc058bfdba.png)
